Added notes for pnpm users when deploying to Netlify (#349)
withastro/docs: src/pages/en/guides/deploy.md

  publish = "dist"
```

> If you are using `pnpm`, then set the following settings instead:

Reviewer: ```suggestion
Using `pnpm`? [Netlify needs some extra set-up to support `pnpm`.](https://answers.netlify.com/t/using-pnpm-and-pnpm-workspaces/2759) We suggest starting with the following settings in your `netlify.toml`:
```

Reviewer 2: ```suggestion
Using [`pnpm` with Netlify?](https://answers.netlify.com/t/using-pnpm-and-pnpm-workspaces/2759) Use the following settings instead:
```

Reviewer: ✂️💨 Scissors to the rescue! LGTM.

Reviewer: Ah, good question. I think I’d say “on”?

Reviewer 2: Updated in my suggestion, thanks! @RafidMuhymin can make the final call.  :)
